wsmatrix@martin.zurowietz.de/IndicatorWsmatrixPopup.js

```diff
@@ -0,0 +1,111 @@
+const { Clutter, GObject, Meta, St } = imports.gi;
+const WsMatrix = imports.misc.extensionUtils.getCurrentExtension();
+const DisplayWrapper = WsMatrix.imports.DisplayWrapper.DisplayWrapper;
+const BaseWorkspaceSwitcherPopup = WsMatrix.imports.BaseWorkspaceSwitcherPopup.BaseWorkspaceSwitcherPopup;
+const WorkspaceSwitcherPopupList = imports.ui.workspaceSwitcherPopup.WorkspaceSwitcherPopupList;
+const Main = imports.ui.main;
+
+var IndicatorWsmatrixPopupList = GObject.registerClass(
+class IndicatorWsmatrixPopupList extends WorkspaceSwitcherPopupList {
+   _init(rows, columns) {
+      super._init();
+      this._rows = rows;
+      this._columns = columns;
+   }
+
+   vfunc_get_preferred_height(forWidth) {
+      let children = this.get_children();
+      let workArea = Main.layoutManager.getWorkAreaForMonitor(Main.layoutManager.primaryIndex);
+      let themeNode = this.get_theme_node();
+
+      let availHeight = workArea.height;
+      availHeight -= themeNode.get_vertical_padding();
+
+      let height = 0;
+      let child = children[0];
+      let [childMinHeight, childNaturalHeight] = child.get_preferred_height(-1);
```

Comment:
`childNaturalHeight` changes from 52 to 50 when switched to the first workspace. This changes the size of all workspace indicators and the whole popup.This needs to be fixed first.

Reply:
will debug this one

Reply:
looks like if the child is the activated workspace, this issue happens. I thought at first it could be a style issue, I also reviewed the default stylesheet of gnome and found that they set the height of the activated workspace, but I had no luck. Also, other GNOME themes seem to have sometimes more noticeable difference and sometimes less noticeable.

I could fix it by taking the maximum height of children[0] or children[1]
this will make sure we always get the same value

```javascript
 let child = children.length <= 1 || children[0].get_preferred_height(-1) > children[1].get_preferred_height(-1) ? children[0] : children[1];
```

Reply:
Here comes another workaround for an unidentified issue. Thanks for debugging!

Reply:
For the record: For me this happens only with the first workspace, no matter if it was active or not.

Reply:
when I made 
```javascript
let child = children[0];
```

index 1 instead of 0, the second workspace was the one causing the issue
```javascript
let child = children[1];
```

so the workspace index that's causing the issue is the active one that has the hardcoded index
that's why I took the max between first and second children

Reply:
I see, you're probably right. Let's see if the GNOME guys have something to say to that. The current workaround should be ok for now.